agent/agent.go

```diff
@@ -1773,15 +1773,20 @@ func (a *agent) Close() error {
 	a.setLifecycle(codersdk.WorkspaceAgentLifecycleShuttingDown)
 
 	// Attempt to gracefully shut down all active SSH connections and
-	// stop accepting new ones.
-	err := a.sshServer.Shutdown(a.hardCtx)
+	// stop accepting new ones. If all processes have not exited after
+	// 10 seconds, we just log it and move on as it's more important
+	// to run the shutdown scripts.
+	sshShutdownCtx, sshShutdownCancel := context.WithTimeout(a.hardCtx, 10*time.Second)
```

Comment:
I can foresee someone wanting to adjust this timeout. Maybe add a CLI option for it?

Reply:
I think that if we do get requests to make this configurable, it will help drive the implementation. I'm not certain this approach is the end-goal we want and I would like to avoid exposing functionality that may limit future implementations.

When we start solving #6175, changing this behavior may be relevant.

I'll go ahead and reduce this to 5 seconds as well as the default docker grace timeout is 10s I believe.

The original motivation behind closing SSH before running shutdown scripts is to ensure that commands doing work exit before potentially doing backups of state or something along those lines. I think giving those commands a grace of 5 seconds is plenty in most cases.